### ISSUE TITLE
Model's forward method and trainer's train step method are incorrectly implemented leading to incorrect loss calculations.

### ISSUE THREAD

**Issue by VecherVhatuX** [VecherVhatuX/learning_data_science_in_prison#348]
The model's forward method is not correctly implemented and the trainer's train step method does not properly clear the model's gradients before each training step, leading to incorrect loss calculations and potentially affecting the model's performance.
